frontend/packages/console-app/src/components/modals/clone/clone-pvc-modal.tsx

```diff
@@ -3,55 +3,69 @@ import './_clone-pvc-modal.scss';
 import {
   LoadingInline,
   ResourceIcon,
   humanizeBinaryBytes,
+  history,
   withHandlePromise,
-} from '@console/internal/components/utils/index';
+  RequestSizeInput,
+  validate,
+  resourceObjPath,
+} from '@console/internal/components/utils';
 import {
   ModalBody,
   ModalComponentProps,
   ModalSubmitFooter,
   ModalTitle,
   createModalLauncher,
 } from '@console/internal/components/factory';
-
-import { DataPoint } from '@console/internal/components/graphs/index';
+import { DataPoint } from '@console/internal/components/graphs';
 import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
-import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
 import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
 import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
-import { getPVCUsedCapacityQuery } from '../../../constants/queries';
 import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { getRequestedPVCSize } from '@console/shared/src/selectors';
 
 const accessModeLabels = Object.freeze({
   ReadWriteOnce: 'Single User (RWO)',
   ReadWriteMany: 'Shared Access (RWX)',
   ReadOnlyMany: 'Read Only (ROX)',
 });
 
+const dropdownUnits = {
+  Mi: 'MiB',
+  Gi: 'GiB',
+  Ti: 'TiB',
+};
+
 const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
   const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
-  const pvcName: string = resource.metadata.name;
-  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+  const { name: pvcName, namespace } = resource?.metadata;
+  const defaultSize: string[] = validate.split(getRequestedPVCSize(resource));
 
-  const pvcUsedCapacityQuery: string = getPVCUsedCapacityQuery(pvcName);
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+  const [requestedSize, setRequestedSize] = React.useState(defaultSize[0] || '');
+  const [requestedUnit, setRequestedUnit] = React.useState(defaultSize[1] || 'Gi');
 
+  const pvcUsedCapacityQuery: string = `kubelet_volume_stats_used_bytes{persistentvolumeclaim='${pvcName}'}`;
   const [response, error, loading] = usePrometheusPoll({
     endpoint: PrometheusEndpoint.QUERY,
     query: pvcUsedCapacityQuery,
-    namespace: resource.metadata.namespace,
+    namespace,
   });
-
   const pvcUsedCapacityQueryResult: DataPoint[] = getInstantVectorStats(
     response,
     null,
     humanizeBinaryBytes,
   );
-
-  const pvcUsedCapacity = pvcUsedCapacityQueryResult?.[0]?.label || 'No Data';
+  const pvcUsedCapacity = pvcUsedCapacityQueryResult?.[0]?.label || '-';
+  const requestedSizeInputChange = ({ value, unit }) => {
+    setRequestedSize(value);
+    setRequestedUnit(unit);
```

Comment:
We shouldn't allow the user to add size lesser than the size of parent PVC, as it will always fail clone creation https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/.

Reply:
@Madhu-1 will you be taking care of it from the backend? 

When clone request is sent for lesser size than the original one can you fail the request? We want to provide the same experience for the CLI user too.

Reply:
Yes the check is present in backend at kubernetes external provisioner sidecar
https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L859
https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L983

Reply:
As I see this it is checking for PVC using the requested size but not the PVC created from `spec.datasource` entity. Also If I am triggering the PVC of smaller size the request is going & PVC is getting created.

Reply:
PVC will get created, but not the PV, PVC creation is under kube-apiserver control not under kubernetes-csi or cephcsi. AFAIK there is no check for this one in kube-apiserver, you may need to disable it in UI for this one.

Reply:
But a clone is a generic feature what if user created it from the CLI? I think we need to fail the API the same way `requested.storage` can't be lesser than the current value.

Also I created the PVC using the clone request the corresponding PV is also getting created.

Reply:
After conversations with @Madhu-1 that Backend is already throwing an error, we don't want to stop user explicitly. @a2batic ^^

Reply:
This can be easily done by passing `minValue` parameter [here](https://github.com/openshift/console/blob/master/frontend/public/components/utils/request-size-input.tsx#L39). It will  help in restricting user from entering lesser value and UI can save one API call in order to realize user of this requirement and here this can easily achieved.

Reply:
Here, you also need to compare units too and so it will be bit of work on UI to do those validations. I am fine with deferring this validation on backend.

Reply:
Correct Kanika & I talked about the min validation but that's not going to work as units are also in place. 
I don't want to go beyond that because this gives different user experience.